### PR TITLE
Various additions to ace_zeus

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Add ability to search players in teleport players UI
tp_ui
- Change group side icons to look better
    - Before:
group_before
    - After:
group_after
- Add Assign Engineer, Assign Repair Vehicle, and Assign Repair Facility modules
- Add Full Heal module
- Improve Attribute Cargo to use `displayName` of objects
    - Before:
cargo_before
    - After:
cargo_after
- Add angle param to function: `getModuleDestination` (fix sideways load into cargo icon)
    - Before: 
loadcargo_before
    - After:
loadcargo_after
- Add Suicide Bomber module
sb_ui

### CODE REVIEW

**Comment by mharis001** [comment]
This PR is done, just needs to be reviewed.

File: addons/zeus/functions/fnc_moduleSetEngineer.sqf [hunk 1]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Author: mharis001
+ * Zeus module function to set unit engineer skill
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSetEngineer.sqf]: Missing final dot.

File: addons/zeus/functions/fnc_moduleSetEngineer.sqf [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ *
+ * Example:
+ * [unit, 2] call ace_zeus_fnc_moduleSetEngineer
+ *
+ * Public: No
+ */
+
+#include "script_component.hpp"
+
+params [["_unit", objNull], ["_engineerN", 1]];
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSetEngineer.sqf]: Is there a case where `_unit` is not passed from ACE3 code? No need to use default value then, rather show an error so we can fix it at source.

File: addons/zeus/functions/fnc_moduleSetEngineer.sqf [hunk 3]
```diff
@@ -0,0 +1,26 @@
+/*
+ * Author: mharis001
+ * Zeus module function to set unit engineer skill
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Skill level (1 - Engineer, 2 - Adv. Engineer) <NUMBER>
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSetEngineer.sqf]: `... <NUMBER> (default: 1)`

File: addons/zeus/functions/fnc_moduleSetRepairFacility.sqf [hunk 4]
```diff
@@ -0,0 +1,51 @@
+/*
+ * Author: mharis001
+ * Assigns object as repair facility
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSetRepairFacility.sqf]: Missing final dot.

File: addons/zeus/functions/fnc_moduleSetRepairVehicle.sqf [hunk 5]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Author: mharis001
+ * Assigns object as repair vehicle
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSetRepairVehicle.sqf]: Missing final dot.

File: addons/zeus/functions/fnc_moduleSuicideBomber.sqf [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+    [QEGVAR(ai,setBehaviour), [[_unit], "CARELESS"], _unit] call CBA_fnc_targetEvent;
+    [QEGVAR(ai,setCombatMode), [_unit, "BLUE"], _unit] call CBA_fnc_targetEvent;
+    [QEGVAR(ai,AISection), [[_unit], ["TARGET", "AUTOTARGET"], false], _unit] call CBA_fnc_targetEvent;
+    [QEGVAR(ai,allowFleeing), [_unit, 0], _unit] call CBA_fnc_targetEvent;
+};
+
+// Run PFH to make unit a suicide bomber
+[{
+    params ["_args", "_pfhID"];
+    _args params [["_unit", objNull], "_activationSide", "_activationRadius", "_explosionSize", "_autoSeek"];
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSuicideBomber.sqf]: This doesn't actually apply the defaults that are listed in header.

File: addons/zeus/functions/fnc_moduleSuicideBomber.sqf [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    // --- Detonation
+    private _nearObjects = (_unit nearObjects _activationRadius) select {side _x == _activationSide && {_x != _unit} && {alive _x}};
+    if !(_nearObjects isEqualTo []) then {
+        createVehicle [EXPLOSIVES select _explosionSize, getPos _unit, [], 0, "CAN_COLLIDE"];
+        [_pfhID] call CBA_fnc_removePerFrameHandler;
+        LOG("Explosion created, PFH removed");
+    };
+
+    // --- Auto Seek
+    if !(_autoSeek) exitWith {};
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSuicideBomber.sqf]: `if (!_autoSeek) ...`

File: addons/zeus/functions/fnc_moduleSuicideBomber.sqf [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+    params ["_args", "_pfhID"];
+    _args params [["_unit", objNull], "_activationSide", "_activationRadius", "_explosionSize", "_autoSeek"];
+
+    // Unit deleted or killed
+    if (isNull _unit || {!alive _unit}) exitWith {
+        [_pfhID] call CBA_fnc_removePerFrameHandler;
+        LOG("Unit deleted or killed, PFH removed");
+    };
+
+    // --- Detonation
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSuicideBomber.sqf]: No `---`.

File: addons/zeus/functions/fnc_moduleSuicideBomber.sqf [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+    private _nearObjects = (_unit nearObjects _activationRadius) select {side _x == _activationSide && {_x != _unit} && {alive _x}};
+    if !(_nearObjects isEqualTo []) then {
+        createVehicle [EXPLOSIVES select _explosionSize, getPos _unit, [], 0, "CAN_COLLIDE"];
+        [_pfhID] call CBA_fnc_removePerFrameHandler;
+        LOG("Explosion created, PFH removed");
+    };
+
+    // --- Auto Seek
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleSuicideBomber.sqf]: No `---`.

File: addons/zeus/functions/fnc_moduleHeal.sqf [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * Return Value:
+ * None
+ *
+ * Example:
+ * [LOGIC] call ace_zeus_fnc_moduleHeal
+ *
+ * Public: No
+ */
+
```
Comment (jonpas) [inline addons/zeus/functions/fnc_moduleHeal.sqf]: No empty line here, here and in other files.

File: addons/zeus/functions/fnc_ui_setEngineer.sqf [hunk 11]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Author: mharis001
+ * Initalizes the `set engineer` zeus module display
```
Comment (jonpas) [inline addons/zeus/functions/fnc_ui_setEngineer.sqf]: `Initalizes the "Set Engineer" Zeus module display.`

File: addons/zeus/functions/fnc_ui_suicideBomber.sqf [hunk 12]
```diff
@@ -0,0 +1,152 @@
+/*
+ * Author: mharis001
+ * Initalizes the `suicide bomber` zeus module display
```
Comment (jonpas) [inline addons/zeus/functions/fnc_ui_suicideBomber.sqf]: `Initalizes the "Suicide Bomber" Zeus module display.`

File: addons/zeus/functions/fnc_ui_suicideBomber.sqf [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+        ["str_a3_cfgvehicles_moduleremotecontrol_f_errorPlayer"] call _fnc_errorAndClose;
+    };
+    case !(_unit getVariable [QGVAR(suicideBomber), false]): {
+        [LSTRING(ModuleSuicideBomber_AlreadyBomber)] call _fnc_errorAndClose;
+    };
+};
+
+// Specific onLoad stuff
+
+// --- Activation side
```
Comment (jonpas) [inline addons/zeus/functions/fnc_ui_suicideBomber.sqf]: No `---` is necessary here.

File: addons/zeus/functions/fnc_ui_suicideBomber.sqf [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+    _ctrl setVariable ["color", _color];
+    _ctrl ctrlSetActiveColor _color;
+
+    _color set [3, 0.5];
+    _ctrl ctrlSetTextColor _color;
+
+    _ctrl ctrlAddEventHandler ["ButtonClick", _fnc_onSelection];
+} forEach SIDE_IDCs;
+
+// --- Activation radius
```
Comment (jonpas) [inline addons/zeus/functions/fnc_ui_suicideBomber.sqf]: No `---` is necessary here.

File: addons/zeus/functions/fnc_ui_suicideBomber.sqf [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+private _edit = _display displayCtrl 83573;
+
+_slider sliderSetRange [5, 50];
+_slider sliderSetSpeed [1, 1];
+_slider sliderSetPosition 10;
+_edit ctrlSetText "10";
+
+_slider ctrlAddEventHandler ["SliderPosChanged", _fnc_onSliderMove];
+
+// --- Confirm and Cancel
```
Comment (jonpas) [inline addons/zeus/functions/fnc_ui_suicideBomber.sqf]: No `---` is necessary here.

File: addons/zeus/functions/fnc_ui_teleportPlayers.sqf [hunk 16]
```diff
@@ -1,5 +1,5 @@
 /*
- * Author: SilentSpike
+ * Author: SilentSpike, mharis001
  * Initalises the `teleport players` zeus module display
```
Comment (jonpas) [inline addons/zeus/functions/fnc_ui_teleportPlayers.sqf]: I know it's not yours but:
`Initalizes the "Teleport Players" Zeus module display.`

**Pull request merged**